pkg/ruler/compat.go

```diff
@@ -2,29 +2,45 @@ package ruler
 
 import (
 	"context"
-	"errors"
 	"time"
 
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/promql"
-	"github.com/prometheus/prometheus/rules"
 	"github.com/prometheus/prometheus/storage"
 	"github.com/weaveworks/common/user"
 
 	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/ruler/rules"
 )
 
+// AppendableHistoryFunc allows creation of an Appendable and AlertHistory to be joined. The default implementation
+// does not take advantage of this option.
+type AppendableHistoryFunc = func(userID string, opts *rules.ManagerOptions) (rules.Appendable, rules.TenantAlertHistory)
+
+// This is the default implementation which returns an unlinked Appendable/AlertHistory.
+func DefaultAppendableHistoryFunc(p Pusher, q storage.Queryable) AppendableHistoryFunc {
```

Comment:
If this function is pluggable, can Loki just inject it's own implementation and Cortex keep the current implementation?